controllers/managedocs_controller_test.go

```diff
@@ -71,8 +71,102 @@ var _ = Describe("ManagedOCS controller", func() {
 			})
 		})
 
+		When("the storeage cluster is not ready", func() {
+			ctx := context.Background()
+			managedOCS := &v1.ManagedOCS{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      ManagedOCSName,
+					Namespace: TestNamespace,
+				},
+			}
+
+			BeforeEach(func() {
```

Comment:
Why do we need this before each when there is only one test it affects? 
It just creates more convoluted code and a discontinuous stack trace  
Is there a specific reason?

Reply:
Yes I wrote it this way to separate the code that sets up the state of the test from the code that tests the behaviour when in that state. Now every `It` within that `When` block can assume the cluster is in a state where:
1. Managed ocs has been created
2. StorageCluster has been created
3. StorageCluster is in a pending Phase